book-content/chapters/08-classes.md

```diff
@@ -0,0 +1,1044 @@
+Classes are a JavaScript feature that help you encapsulate data and behavior into a single unit. They are a fundamental part of object-oriented programming and are used to create objects that have properties and methods.
+
+You can use the `class` keyword to define a class, and then create instances of that class using the `new` keyword. TypeScript adds a layer of static type checking to classes, which can help you catch errors and enforce structure in your code.
+
+Let's build a class from scratch and see how it works.
+
+## Creating a Class
+
+To create a class, you use the `class` keyword followed by the name of the class. Similar to types and interfaces, the convention is to have the name in PascalCase, which means the first letter of each word in the name is capitalized.
+
+We'll start creating the `Album` class in a similar way to how a type or interface is created:
+
+```ts twoslash
+// @errors: 2564
+class Album {
+  title: string;
+  artist: string;
+  releaseYear: number;
+}
+```
+
+At this point, even though it looks like a type or interface, TypeScript gives an error for each property in the class.
+How do we fix this?
+
+### Adding a Constructor
+
+In order to fix these errors, we need to add a `constructor` to the class. The `constructor` is a special method that runs when a new instance of the class is created. It's where you can set up the initial state of the object.
+
+To start, we'll add a constructor that assigns values for the properties of the `Album` class:
+
+```typescript
+class Album {
+  title: string;
+  artist: string;
+  releaseYear: number;
+
+  constructor() {
+    this.title = "Loop Finding Jazz Records";
+    this.artist = "Jan Jelinek";
+    this.releaseYear = 2001;
+  }
+}
+```
+
+Now, when we create a new instance of the `Album` class, we can access the properties and values we've set in the constructor:
+
+```typescript
+const loopFindingJazzRecords = new Album();
+
+console.log(loopFindingJazzRecords.title); // Output: Loop Finding Jazz Records
+```
+
+The `new` keyword creates a new instance of the `Album` class, and the constructor sets the initial values of our class's properties. In this case, because the properties are hardcoded, every instance of the `Album` class will have the same values.
+
+#### You Don't Always Need To Type Class Properties
+
+As we'll see, TypeScript can do some really smart inference with classes. It's able to infer the types of the properties from where we assign them in the constructor, so we can actually drop some of the type annotations:
+
+```typescript
+class Album {
+  title;
+  artist;
+  releaseYear;
+
+  constructor() {
+    this.title = "Loop Finding Jazz Records";
+    this.artist = "Jan Jelinek";
+    this.releaseYear = 2001;
+  }
+}
```

Comment:
It's simplifiable even further, as long as we don't need to depend on any arguments:
```ts
class Album {
  title = "Loop Finding Jazz Records";
  artist = "Jan Jelinek";
  releaseYear = 2001;
}
```